lib/appmap/handler/rails/sql_handler.rb

```diff
@@ -132,16 +132,42 @@ def database_type
             end
 
             def in_transaction?
-              ActiveRecord::Base.connection.open_transactions > 0
+              execute_in_context do
+                ActiveRecord::Base.connection_pool.with_connection do
+                  ActiveRecord::Base.connection.open_transactions > 0
+                end
+              end
             end
 
             def execute_query(sql)
-              ActiveRecord::Base.connection.execute(sql).to_a
+              execute_in_context do
+                ActiveRecord::Base.connection_pool.with_connection do
+                  ActiveRecord::Base.connection.execute(sql).to_a
+                end
+              end
+            end
+
+            private
+
+            # Assure Thread and Fiber safety for ActiveRecord queries.
+            # This is only necessary for applications using graphql-ruby.
+            # This change wraps the
+            def execute_in_context
```

Comment:
I believe [with_connection](https://github.com/rails/rails/pull/44219/files#diff-642b90553b888bd2c724c093a1a685a5408a7d8293f3751366c25dc548936eb7R210) already handles thread/fiber-safety, so I'm not sure if this conditional is required. Are you certain that it is?

Reply:
@swrobel it does, except in AppMap's case. I updated the description to explain why the wrapped Fiber is needed.